#!/usr/bin/env python
"""Parameters for request sent to sonar.
"""

PACKAGE = "omniscan_sidescan"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("speed_of_sound_mm", int_t, 0, "speed of sound in water", 1482000, 1482000, 1500000)
gen.add("start_mm", int_t, 0, "start of ping range, in mm", 0, 0, 5000)
gen.add("length_mm", int_t, 0, "length of returned profile, end of region = start_mm + length_mm", 5000, 0, 5000)
gen.add("msec_per_ping", int_t, 0, "set value to limit ping rate; set to 0 for best ping rate", 0, 10)
gen.add("pulse_len_percent", double_t, 0, "% of total ping time for current range", 0.002, 0.002, 0.004)
gen.add("filter_duration_percent", double_t, 0, "typically 0.0015", 0.0015, 0.0015, 0.0016)
gen.add("gain_index", int_t, 0, "set to -1 for auto gain, otherwise 0-7 sets gain", -1, -1, 7)
gen.add("num_results", int_t, 0, "number of signal data points in returning profile", 600, 200, 1200)

exit(gen.generate(PACKAGE, "omniscan_sidescan", "Omniscan450"))